File: client/src/models/ZahtjevZaPregled.ts
import type { Doktor } from "./Doktor";
import type { Pacijent } from "./Pacijent";

export type ZahtjevZaPregled = {
  id: number;
  pacijentId: number;
  pacijent: Pacijent;
  doktorId: number;
  doktor: Doktor;
  doktorIme: string;
  doktorPrezime: string;
  pacijentIme: string;
  pacijentPrezime: string;
  datumZahtjeva: string; 
  opis: string;
  status: "Na čekanju" | "Prihvaćen" | "Odbijen"; 
};

File: API/Entities/ZahtjevZaPregled.cs
using API.Entities;

public class ZahtjevZaPregled
{
    public int Id { get; set; }

    public int PacijentId { get; set; }
    public required Pacijent Pacijent { get; set; }

    public int DoktorId { get; set; }
    public required Doktor Doktor { get; set; }

    public DateTime DatumZahtjeva { get; set; } = DateTime.Now;

    public required string Opis { get; set; }

    public string Status { get; set; } = "Na čekanju"; 
}
